pkg/volume/util/resize_util.go

```diff
@@ -54,16 +56,37 @@ func MarkFSResizeFinished(
 	newPVC := pvc.DeepCopy()
 	newPVC.Status.Capacity = capacity
 	newPVC = MergeResizeConditionOnPVC(newPVC, []v1.PersistentVolumeClaimCondition{})
-	_, err := PatchPVCStatus(pvc /*oldPVC*/, newPVC, kubeClient)
+	_, err := PatchPVCStatus(pvc /*oldPVC*/, newPVC, kubeClient, false)
 	return err
 }
 
 // PatchPVCStatus updates PVC status using PATCH verb
+// If the PVC was requed for processing then we MUST add version check to PATCH
+// request so as to ensure that we aren't working with stale PVC objects.
```

Comment:
If the version check fails and assuming PVC still needs a resize, then it will be queued again for resize. But if there was a race with kubelet and resize was finished on kubelet - then the PVC won't be queued again for resize (and hence we will not have erroneous condition added to PVC anymore).

> Why don't we always do versionCheck? What if PVC is modified by some outside component?
While PVC indeed can be modified by other components there are only 2 components that can modify `Status.Capacity` and `Status.Conditions` - either Kubelet or expand_controller (or an external expand_controller when it exists). When more size is requested by the user for the first time, PVC can only be modified by controller before kubelet acts on it.  Other modifications to the PVC can happen regardless as long as they don't modify these 2 fields. 

Having said that - we indeed can introduce version check for all cases and since there is already an reque logic - it should work. But - I did not want PVC resize to reque for unrelated modifications and secondly since there is no race in other cases I did not think it is worth fixing what isn't broken.

Reply:
I would argue that PVC is user object and they can modify it at any time, presumably "fixing" things. I'd be rather on the safe side and either use full `Update` or always patch with resource version in this case.

Reply:
So the one downside of *always* adding resource verison check is - it could potentially delay starting the pod because resizing is done as part of mounting process and if updating PVC failed then the entire operation has to be requeued. 

This function also gets called from kubelet.